executor/simple.go

```diff
@@ -1225,7 +1225,7 @@ func (e *SimpleExec) executeDropUser(ctx context.Context, s *ast.DropUserStmt) e
 
 		// begin a transaction to delete a user.
 		sql.Reset()
-		sqlexec.MustFormatSQL(sql, `DELETE FROM %n.%n WHERE Host = %? and User = %?;`, mysql.SystemDB, mysql.UserTable, user.Hostname, user.Username)
+		sqlexec.MustFormatSQL(sql, `DELETE FROM %n.%n WHERE Host = %? and User = %?;`, mysql.SystemDB, mysql.UserTable, strings.ToLower(user.Hostname), user.Username)
```

Comment:
I think there might be another case in line 680:
```golang
sqlexec.MustFormatSQL(sql, `DELETE IGNORE FROM %n.%n WHERE FROM_HOST=%? and FROM_USER=%? and TO_HOST=%? and TO_USER=%?`, mysql.SystemDB, mysql.RoleEdgeTable, role.Hostname, role.Username, user.Hostname, user.Username)
```

Reply:
I'm not sure if it needs to be modified. Because this is not the `mysql.user table`, currently, `create user` will lowercase `hostname`, so I think all SQL involving `mysql.user` `hostname` should be changed to lowercase. I'm not sure about the other tables.

I added some other modifications, please help double confirm whether it is correct, thank you.

@mjonss